### PR TITLE
Python module structure

### LINKED ISSUE #2122: Avoid issues with imports within Python module
There are a couple of problems with the current way we import things within the scipp Python module:

- `from .._scipp import core as sc` is used in many places to get access to "core" scipp functionality
  - This is risky since this essentially "ignores" the Python wrappers of these functions. It also does not give access to some Python-only functionality.
- `from .. import xyz` is used in some places, to either avoid the former, or avoid specifying a precise name

I feel the main part of the problem is that we don't have a way to import the "core" part (*including* Python wrappers) of scipp in any file that is not part of the "core", such as the various visualization bits, or the I/O code. I am not sure what the solution is. One idea is:
- Move the import of `.._scipp` from the top-level `__init__.py` into a `core/` submodule.
- Move all the wrappers into `core/`
- Import what is needed in the top-level `__init__.py` from `core/`
- Non-core functionality can use `from ..core import xyz` or `import ..core as sc` (is that valid?)

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Fixes #2122.

### CODE REVIEW

File: python/src/scipp/__init__.py [hunk 1]
```diff
@@ -26,32 +26,19 @@
 config = runtime_config.load()
```
Comment (nvaytet) [inline python/src/scipp/__init__.py:26]: I think it's looking good so far. :+1:

File: python/src/scipp/__init__.py [hunk 2]
```diff
@@ -26,32 +26,19 @@
...
                          DataArrayError, DatasetError, DimensionError, \
                          DTypeError, NotFoundError, SizeError, SliceError, \
                          UnitError, VariableError, VariancesError
 # Import submodules
-from ._scipp.core import units, dtype
+from .core import units, dtype
 from . import geometry
 # Import functions
-from ._scipp.core import as_const, choose, logical_and, logical_or, logical_xor, where
+from ._scipp.core import as_const, choose, logical_and, logical_or, logical_xor
```
Comment (jl-wynen) [inline python/src/scipp/__init__.py:41]: Is there a reason why those are imported from `_scipp.core` directly and not through the `core` subpackage like the rest?

File: python/src/scipp/__init__.py [hunk 3]
```diff
@@ -72,61 +58,15 @@ def custom_formatwarning(msg, *args, **kwargs):
...
-# Prevent unwanted conversion to numpy arrays by operations. Properly defining
-# __array_ufunc__ should be possible by converting non-scipp arguments to
-# variables. The most difficult part is probably mapping the ufunc to scipp
-# functions.
-for _cls in (Variable, DataArray, Dataset):
-    setattr(_cls, '__array_ufunc__', None)
-del _cls
+from .core.variable import scalar, zeros, zeros_like, ones, ones_like, empty, \
+        empty_like, full, full_like, matrix, matrices, vector, vectors, array, \
+        linspace, geomspace, logspace, arange
```
Comment (jl-wynen) [inline python/src/scipp/__init__.py:69]: Are the other `_xxx` modules also supposed to go into the core subpackage?
If not, why is `variable` in there but not `_dataset`?

File: python/src/scipp/__init__.py (hunk 2 above)
Comment (SimonHeybrock) [reply python/src/scipp/__init__.py:41]: No, just incomplete. Fixed by now.

File: python/src/scipp/__init__.py (hunk 3 above)
Comment (SimonHeybrock) [reply python/src/scipp/__init__.py:69]: Yes they will, this is WIP.

**Review by OwenArnold — COMMENTED** [review]
Looks good.

**Pull request merged**